rs/plotting/src/plotter_gui/mod.rs

```diff
@@ -165,127 +170,211 @@ impl eframe::App for PlotterGuiState {
+                                        }
+
+                                        if y > &data_driven_largest_y {
+                                            data_driven_largest_y = *y;
+                                        }
+
+                                        points.push(egui_plot::PlotPoint::new(*x, *y));
                                     }
-                                    let plot_points = egui::plot::PlotPoints::Owned(points);
+                                    let plot_points = egui_plot::PlotPoints::Owned(points);
 
                                     match g.curve_type {
                                         crate::graphs::common::LineType::LineStrip => {
                                             plot_ui.line(
-                                                egui::plot::Line::new(plot_points)
+                                                egui_plot::Line::new(plot_points)
                                                     .color(g.color)
                                                     .name(curve_name),
                                             );
                                         }
                                         crate::graphs::common::LineType::Points => {
                                             plot_ui.points(
-                                                egui::plot::Points::new(plot_points)
+                                                egui_plot::Points::new(plot_points)
                                                     .color(g.color)
                                                     .name(curve_name),
                                             );
                                         }
                                     }
                                 }
                                 GraphType::Vec3(g) => {
-                                    let mut points = vec![];
-                                    points.push(vec![]);
-                                    points.push(vec![]);
-                                    points.push(vec![]);
+                                    let mut points = vec![vec![], vec![], vec![]];
 
                                     for (x, y) in &g.data {
-                                        points[0].push(egui::plot::PlotPoint::new(*x, y.0));
-                                        points[1].push(egui::plot::PlotPoint::new(*x, y.1));
-                                        points[2].push(egui::plot::PlotPoint::new(*x, y.2));
+                                        if x > &data_driven_largest_x {
+                                            data_driven_largest_x = *x;
+                                        }
+
+                                        let max_y = y.0.max(y.1.max(y.2));
+                                        if max_y > data_driven_largest_y {
+                                            data_driven_largest_y = max_y;
+                                        }
```

Comment:
I have a vague sense that we're interleaving UI code and UI model code more than would be ideal, at least in this `.show()` lambda, but I'm not very familiar with this area of the code.

Reply:
Good question. Ideally we have more general predicates and some sort of DSL, so we can formulate:

```
x-max-bound = MAX_X(DATA)
```

But I do not think it is worth the time working on something like this now, even though it would be cleaner.